addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -265,9 +268,47 @@ def _process_pickings(self):
         if unsatisfied:
             # Unreserve if the picking type cannot handle partials or it
             # can but there is nothing allocated (no stock.move.lines)
-            if not self.picking_type_id.u_handle_partials or not mls:
+            # If there are several picking types, they should have all same handle_partials,
+            # can be improved to handle it better if necessary.
+            handle_partial = self.picking_type_id[0].u_handle_partials
+            if not handle_partial or not mls:
                 # construct error message, report only products
                 # that are unreservable.
                 moves = unsatisfied.move_lines.filtered(unsatisfied_state)
                 self.raise_insufficiency_error(moves)
         return {"newly_processed": newly_processed, "mls": mls}
+
+    def add_more_pickings_to_reserve(self, picking_type):
+        """
+        Checking different configuration of the picking type, and add more pickings to reserve when applied.
+        Method can be override on specific implementations to extend pickings to reserve.
+
+        Args:
+            picking_type (recordset): Stock picking type recordset.
+        Returns:
+            pickings (recordset): Picking to reserve recordset
+        """
+        pickings = self
+        if self.batch_id:
+
+            if self.u_reserve_batches:
```

Comment:
nit: we should combine these branches unless we plan on changing behaviour of one over the other in future
```
if self.u_reserve_batches or picking_type.u_reserve_batches:
    pickings = pickings.batch_id.picking_ids
```